### PR TITLE
rustdoc: Add more test coverage

### PR DESCRIPTION
**Pull request by onlineSoftwareDevOK**
Related issue https://github.com/rust-lang/rust/issues/91113

### CODE REVIEW

File: src/test/rustdoc/nested-modules.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#![crate_name = "foo"]
+
+mod a_module {
+  fn private_function() {
+      println!("called `a_module::private_function()`");
```
Comment (camelid) [inline src/test/rustdoc/nested-modules.rs:5]: Rustdoc ignores the bodies of functions, so can you remove the `println!`s here and below?

```suggestion
```

File: src/test/rustdoc/nested-modules.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  }
+
```
Comment (camelid) [inline src/test/rustdoc/nested-modules.rs:7]: (You may have to adjust the test until it passes.)

```suggestion

// @has 'foo/a_module/index.html' '//a[@href="fn.private_function.html"]' 'other_private_function'
use private_function as other_private_function;

```

File: src/test/rustdoc/nested-modules.rs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  pub mod a_nested_module {
+    pub fn a_nested_public_function() {
+        println!("called `a_module::a_nested_module::a_nested_public_function()`");
+    }
```
Comment (camelid) [inline src/test/rustdoc/nested-modules.rs:11]: (Fill in the '...' with the relevant info.)

```suggestion
    }

    // @has 'foo/a_nested_module/index.html' '//a[@href="..."]' 'pub use ...'
    pub use a_nested_public_function as another_nested_public_function;
```

File: src/test/rustdoc/nested-modules.rs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  }
+}
```
Comment (camelid) [inline src/test/rustdoc/nested-modules.rs:13]:
```suggestion

    // @has ...
    pub use a_nested_module::a_nested_public_function as a_nested_module::yet_another_nested_public_function;
    // @has ...
    pub use a_nested_module::another_nested_public_function as one_last_nested_public_function;

}
```

File: src/test/rustdoc/nested-modules.rs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+// @!has foo/all.html 'a_module'
+// @has foo/all.html 'a_nested_module'
```
Comment (camelid) [inline src/test/rustdoc/nested-modules.rs:16]: Can you change these to use `foo/index.html` and `//a[@href` like above?

File: src/test/rustdoc/nested-modules.rs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+pub use a_module::a_nested_module;
```
Comment (camelid) [inline src/test/rustdoc/nested-modules.rs:17]:
```suggestion
pub use a_module::a_nested_module;

// @has ...
// @has ...
// ...
pub use a_module::{a_nested_module::{a_nested_public_function, another_nested_public_function}, yet_another_nested_public_function, one_last_nested_public_function};

// @has ...
// @has ...
pub use a_module::{private_function, another_private_function};
```

**Review by camelid — COMMENTED** [review]
Hi @onlineSoftwareDevOK, thanks for your PR!

I was more thinking that this would test re-exporting `private_function` `a_nested_public_function` in `a_module` and then in the top-level namespace.

**_However_**, this test seems useful as well! I left suggestions of how to extend it so it can test some more things too.

Also, could you run `rustfmt` on this file to fix the indentation? (We use 4 spaces in this repo.)

Feel free to ask for clarification on my comments.

**Comment by camelid** [comment]
By the way, when you're ready for more feedback, please post a comment with the content `@rustbot ready`. This will mark the PR as ready for another review.

File: src/test/rustdoc/nested-modules.rs (hunk 1 above)
Comment (onlineSoftwareDevOK) [reply src/test/rustdoc/nested-modules.rs]: Hi @camelid, thanks for your didactic feedback!

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks for working on this! Looks good to me. Do you see anything else @camelid?

File: src/test/rustdoc/nested-modules.rs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    // @!has aCrate/a_nested_module/index.html 'yet_another_nested_public_function'
+    pub use a_nested_module::a_nested_public_function as yet_another_nested_public_function;
+
+    // @!has aCrate/a_nested_module/index.html 'one_last_nested_public_function'
+    pub use a_nested_module::another_nested_public_function as one_last_nested_public_function;
+}
+
+//has aCrate/index.html aCrate
```
Comment (GuillaumeGomez) [inline src/test/rustdoc/nested-modules.rs:25]: This comment isn't testing anything. Better remove it.

Comment (onlineSoftwareDevOK) [reply src/test/rustdoc/nested-modules.rs]: Done. Thanks for reviewing my first contribution.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by onlineSoftwareDevOK** [comment]
> Thanks! Please squash your commits then I'll approve it.

Done. Thanks in advance!

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks again!

@bors r+ rollup

**Pull request merged**